src/runtime/auto-prompt-manager.js

```diff
@@ -0,0 +1,172 @@
+   * A prompt may not be displayed if the appropriate criteria are not met.
+   * @param {{
+   *   autoPromptType: (string|undefined),
+   *   alwaysShow: (boolean|undefined),
+   * }=} options
+   * @param {function()=} displayForLockedContentFn
+   * @return {!Promise}
+   */
+  showAutoPrompt(options, displayForLockedContentFn) {
+    // Manual override of display rules, mainly for demo purposes.
+    if (options.alwaysShow) {
+      // TODO(stellachui): Show the mini prompt.
+      return Promise.resolve();
+    }
+
+    // Fetch entitlements and the client config from the server, so that we have
+    // the information we need to determine whether and which prompt should be
+    // displayed.
+    return Promise.all([
+      this.clientConfigManager_.getAutoPromptConfig(),
+      this.entitlementsManager_.getEntitlements(),
+    ]).then((values) => {
+      this.showAutoPrompt_(
+        values[0],
+        values[1],
+        options,
+        displayForLockedContentFn
+      );
+    });
+  }
+
+  /**
+   * Displays the appropriate auto prompt, depending on the fetched prompt
+   * configuration, entitlement state, and options specified.
+   * @param {!../model/auto-prompt-config.AutoPromptConfig} autoPromptConfig
+   * @param {!../api/entitlements.Entitlements} entitlements
+   * @param {{
+   *   autoPromptType: (string|undefined),
+   *   alwaysShow: (boolean|undefined),
+   * }=} options
+   * @param {function()=} displayForLockedContentFn
+   */
+  showAutoPrompt_(
+    autoPromptConfig,
+    entitlements,
+    options,
+    displayForLockedContentFn
+  ) {
+    if (!this.shouldShowMiniPrompt_(autoPromptConfig, entitlements, options)) {
+      if (
+        this.shouldShowLockedContentPrompt_(entitlements) &&
+        displayForLockedContentFn
+      ) {
+        displayForLockedContentFn();
+      }
+      return;
+    }
+    // TODO(stellachui): Show the mini prompt.
+  }
+
+  /**
+   * Determines whether a mini prompt for contributions or subscriptions should
+   * be shown.
+   * @param {!../model/auto-prompt-config.AutoPromptConfig} autoPromptConfig
+   * @param {!../api/entitlements.Entitlements} entitlements
+   * @param {{
+   *   autoPromptType: (string|undefined),
+   *   alwaysShow: (boolean|undefined),
+   * }=} options
+   * @returns {boolean}
+   */
+  shouldShowMiniPrompt_(autoPromptConfig, entitlements, options) {
+    // If the mini auto prompt type is not supported, don't show the prompt.
+    if (
+      options.autoPromptType === undefined ||
+      options.autoPromptType == AutoPromptType.NONE
+    ) {
+      return false;
+    }
+
+    // If we found a valid entitlement, don't show the prompt.
+    if (entitlements.enablesThis()) {
+      return false;
+    }
+
+    // The mini auto prompt is only for non-paygated content.
+    if (this.pageConfig_.isLocked()) {
+      return false;
+    }
+
+    // Don't cap subscription prompts.
+    if (options.autoPromptType == AutoPromptType.SUBSCRIPTION) {
```

Comment:
Could you use the triple equals whenever possible? It's more predictable
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

Reply:
Done.